controllers/usersignup/status_updater.go

```diff
@@ -440,3 +440,8 @@ func (u *StatusUpdater) updateStatusConditions(ctx context.Context, userSignup *
 	}
 	return u.Client.Status().Update(ctx, userSignup)
 }
+
+func (u *StatusUpdater) updateStatusHomeSpace(ctx context.Context, userSignup *toolchainv1alpha1.UserSignup, spaceName string) error {
+	userSignup.Status.HomeSpace = spaceName
+	return u.Client.Status().Update(ctx, userSignup)
```

Comment:
I'm not sure if it's needed and maybe the client already takes care of it but just to be on the safe side:

```suggestion
	if userSignup.Status.HomeSpace != spaceName {
		userSignup.Status.HomeSpace = spaceName
		return u.Client.Status().Update(ctx, userSignup)
	}
	return nil // Nothing changed
```